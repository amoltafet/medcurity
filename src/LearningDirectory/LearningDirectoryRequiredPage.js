import React from 'react';
import './LearningDirectoryPage.css';
import 'bootstrap/dist/css/bootstrap.min.css'
import { useEffect, useState } from "react";
import Axios from 'axios';
import { LearningDirectoryPageContent } from './LearningDirectoryPage';

const LearningDirectoryRequiredPage = () => {
    const [learningModules, setLearningModules] = useState([])
    Axios.defaults.withCredentials = true;
    const [session, setSession] = useState([]);
    const [isLoading, setLoading] = useState(true)

    useEffect(() => {
        Axios.get("http://localhost:3002/users/login").then((response) => {
          console.log('aaahhh', response.data.user)
          setSession(response.data.user[0])
        });
      }, []);

    useEffect(() => {
        console.log("testing")
        if (session.user != undefined) {
            setLoading(false)
            console.log("changed")
        }
    }, [session])

    

    // Query for getting user's required learning modules
    useEffect(() => {
        console.log("ran")
        if (!isLoading) {
            console.log(String(session.user.userid))
            Axios.get('http://localhost:3002/api/getQuery', 
                { params: { the_query: 'SELECT * ' +
                'FROM LearningModules JOIN AssignedLearningModules ON LearningModules.ID = AssignedLearningModules.LearningModID ' +
                'WHERE AssignedLearningModules.UserID = ' + String(session.user.userid)} 
                }).then((response) => {
                    setLearningModules(Object.values(response.data))
            });
            
        }
    }, [isLoading])


    return(
        <LearningDirectoryPageContent directoryTitle="Required Modules Directory" modules={learningModules} />
    );

}

export default LearningDirectoryRequiredPage;