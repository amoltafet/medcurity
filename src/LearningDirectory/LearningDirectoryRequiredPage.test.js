import renderer from 'react-test-renderer';
import LearningDirectoryRequiredPage from './LearningDirectoryRequiredPage';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';

test('renders LoginPage', () => {
  const tree = renderer.create(<BrowserRouter><LearningDirectoryRequiredPage /></BrowserRouter>).toJSON();
  expect(tree).toMatchSnapshot();
});