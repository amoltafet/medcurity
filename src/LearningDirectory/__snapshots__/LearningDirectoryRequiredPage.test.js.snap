// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LoginPage 1`] = `
Array [
  <div
    className=" LearningDirectoryPageContainer container"
  >
    <h1>
      Required Modules Directory
    </h1>
  </div>,
  <div
    className="dashboard card-deck"
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
      }
    }
  />,
  <div
    className="d-grid gap-2"
  />,
]
`;
